main.tf

```diff
@@ -43,13 +43,46 @@ resource "aws_key_pair" "auth" {
 
 resource "aws_s3_bucket" "acme_main" {
   bucket = "test-bucket"
-  acl    = "public-read"
+  acl    = "private"
   versioning {
-    enabled = false
+    enabled    = true
     mfa_delete = true
   }
   website {
     index_document = "index.html"
     error_document = "error.html"
   }
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+}
+resource "aws_s3_bucket_policy" "acme_main" {
+  bucket = "${aws_s3_bucket.acme_main.id}"
+
+  policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "acme_main-restrict-access-to-users-or-roles",
+      "Effect": "Allow",
+      "Principal": [
+        {
+          "AWS": [
+            arn:aws:iam::641885301384:role/Accurics_Read_Only_Role,
+            arn:aws:iam::641885301384:user/harshit2
+          ]
+        }
+      ],
+      "Action": "s3:GetObject",
+      "Resource": "arn:aws:s3:::${aws_s3_bucket.acme_main.id}/*"
+    }
```

Comment:
**Accurics remediation**
 
**[HIGH]: S3 bucket should not be world-listable**
 
**Vulnerability**
Misconfigured S3 buckets can leak private information to the entire internet or allow unauthorized data tampering / deletion
 
**Remediation**
In the S3 console, select the Permissions tab, and then Bucket Policy. Remove policies for s3:List* actions for any principals '*'. If necessary, modify the policy instead, to limit the access to specific principals. Then, select Access Control List, and remove any grants to List objects to Public (everyone) users. Refer to: https://docs.aws.amazon.com/AmazonS3/latest/dev/using-iam-policies.html